tests/test_dataloader.py

```diff
@@ -394,6 +394,50 @@ def test_finetuning_dataloader_small_data(dataset_size: int,
         shutil.rmtree(tiny_dataset_folder_path)
 
```

Comment:
Can we add a test for the build_hf_dataset_from_remote function?

Reply:
I think you can call `_build_hf_dataset_from_remote` with get_file mocked to call `make_tiny_ft_dataset`

Reply:
Or alternatively, we can refactor so that we have a separate function that creates the paths (like finetune_dir, etc.) for`_build_hf_dataset_from_remote`. We can then call that function in a unit test and check to see that the paths created are as expected.

Reply:
added